save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DemoManagerController.kt

```diff
@@ -97,16 +94,20 @@ class DemoManagerController(
                 }
             }
         }
-        .switchIfEmptyToResponseException(HttpStatus.CONFLICT) {
-            // todo: will be removed when uploading files will be implemented
-            "Right now save-demo requires github repository to download tool. Please provide github repository."
-        }
         .flatMap {
             webClientDemo.post()
                 .uri("/demo/internal/add-tool")
                 .contentType(MediaType.APPLICATION_JSON)
                 .bodyValue(demoDto)
                 .retrieve()
+                .onStatus({ it == HttpStatus.CONFLICT }) {
+                    Mono.error(
+                        ResponseStatusException(
+                            HttpStatus.CONFLICT,
+                            "Demo creation request is invalid: fill project coordinates, run command and file name.",
```

Comment:
Shouldn't it be `400 bad request`?

Reply:
From docs

409 Conflict
Indicates that the request could not be processed because of conflict in the current state of the resource, **such as an between multiple simultaneous updates**

Reply:
Oh got it. We used to return `Conflict` when data is invalid.